### ISSUE TITLE
control flow migration: ngIf else error

### ISSUE THREAD

**Issue by e-oz** [angular/angular#53113]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

Was:
```html
<span *ngIf="instance.name else unnamed">{{instance.name}}</span>

<ng-template #unnamed>
    <span i18n>Unnamed Instance</span>
  </ng-template>
```

After migration:
```html
@if (b.name else unnamed) {
  <span>{{b.name}}</span>
}

<ng-template #unnamed>
    <span i18n>Unnamed Instance</span>
  </ng-template>
```

it gives error: `NG5002: Parser Error: Unexpected token 'else'`

### Please provide the exception or error you saw

```true
NG5002: Parser Error: Unexpected token 'else'
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
"@angular/core": "^17.0.4"
```

### Anything else?

I have to say, that this migration right now works much better - thank you for all bugfixes!

**Comment by e-oz** [comment]
After fixing cases like this manually, a few observations:
* Sometimes it is possible to replace such cases by inserting a template into the `@else` block, but sometimes, when the template is being used more than in one place, it's better to leave the template and use `<ng-container *ngTemplateOutlet>`;
* The most difficult case is nested `ngIf else template`.

**Comment by jessicajaniuk** [comment]
@e-oz Thanks for the report! You found this error before my PR got approved / merged. It'll be fixed in #53076. The main issue here is that typically we expect `else` statements to be preceded by a `;`. The migration hadn't accounted for it being omitted, which as it turns out is valid syntax. And yeah, your described situation is exactly the case I handled. It will remove the template outlet and template in the case that there's only one usage.

**Comment by jessicajaniuk** [comment]
#53076 has now been merged. Marking this as fixed. Feel free to try it out when 17.0.0.5 releases tomorrow.
